tests/unit/pluginhandler/test_pluginhandler.py

```diff
@@ -1807,6 +1807,25 @@ def test_clean_prime_state(self):
         self.assertThat(self.handler.next_step(), Equals(steps.PRIME))
         self.assertFalse(os.path.exists(bindir))
 
+    def test_clean_prime_state_inconsistent_files(self):
+        self.assertRaises(errors.NoLatestStepError, self.handler.latest_step)
+        self.assertThat(self.handler.next_step(), Equals(steps.PULL))
+        bindir = os.path.join(self.prime_dir, "bin")
+        os.makedirs(bindir)
+        open(os.path.join(bindir, "1"), "w").close()
+
+        self.handler.mark_done(steps.STAGE)
+
+        self.handler.mark_done(
```

Comment:
mind adding a comment about adding a file bin/2 and dir foo as primed but having them non existent?
Alternatively, invert the order and add a comment to the file creation part
```diff
diff --git a/tests/unit/pluginhandler/test_pluginhandler.py b/tests/unit/pluginhandler/test_pluginhandler.py
index 930ea82..4ceda2c 100644
--- a/tests/unit/pluginhandler/test_pluginhandler.py
+++ b/tests/unit/pluginhandler/test_pluginhandler.py
@@ -1810,9 +1810,6 @@ class StateTestCase(StateBaseTestCase):
     def test_clean_prime_state_inconsistent_files(self):
         self.assertRaises(errors.NoLatestStepError, self.handler.latest_step)
         self.assertThat(self.handler.next_step(), Equals(steps.PULL))
-        bindir = os.path.join(self.prime_dir, "bin")
-        os.makedirs(bindir)
-        open(os.path.join(bindir, "1"), "w").close()
 
         self.handler.mark_done(steps.STAGE)
 
@@ -1820,6 +1817,11 @@ class StateTestCase(StateBaseTestCase):
             steps.PRIME, states.PrimeState({"bin/1", "bin/2"}, {"bin", "foo"})
         )
 
+        # Only create a subset of the files and directories that were marked as primed
+        bindir = os.path.join(self.prime_dir, "bin")
+        os.makedirs(bindir)
+        open(os.path.join(bindir, "1"), "w").close()
+
         self.handler.clean_prime({})
 
         self.assertThat(self.handler.latest_step(), Equals(steps.STAGE))
```